### ISSUE TITLE
Backspace - Gurmukhi keyboard adds undefined to text on click

### ISSUE THREAD

**Issue by harpreetkhalsagtbit** [KhalisFoundation/SikhiToTheMax-Web#424]
## What went wrong ?
![stm](https://user-images.githubusercontent.com/2958292/38869403-3dfb84da-4268-11e8-9de7-2e6ba37c2d1c.png)

Using gurmukhi keyboard for searching, if you input some character and then click on backspace, it adds undefined to othe searchtext.

## How can authors of this project cause the issue to occur on their systems ?

**Case 1**

  1. click on punjabi character
  2. click on backspace icon
  3. text will become - `bundefined` (in punjabi)

**Case 2**

  1. click on backspace icon
  2. text will become - `undefined` (in punjabi)

## What were you trying the project on ?

  - Browser: Chrome
  - Browser Version: 65
  - Operating System: Ubuntu

##My Effort
I tried to fix this issue. What I found is

```js
    const $button =
      _$button.nodeName.toUpperCase() === 'I' ? _$button.parentNode : _$button;
```
This condition is falsy for backspace button 
` _$button.nodeName.toUpperCase() returns `SVG` not `I``

I wanted to fix that but as I am new to this project, just today I have looked at this project, I dont know what this condition actually does. Why you are comparing it with 'I'. If you can explain then I can give a better solution and will send a P.R

Thanks, I really want to contribute in this project. I can help you guys in many ways.
@navdeepsinghkhalsa  @ManjotS @bogas04

**Comment by bogas04** [comment]
Thank you so much for filing the bug.

We were using event delegation when the codebase was written in jQuery, however React does it automatically for all its event listeners, so we need not optimize for that.

I've fixed this issue and #246 by rewriting the code in slightly better way.

I would request you to review it once :)

https://github.com/KhalisFoundation/SikhiToTheMax-Web/pull/425

**Comment by bogas04** [comment]
Please test on http://dev.sikhitothemax.org
